src/hardware/adc.rs

```diff
@@ -148,7 +204,38 @@ impl Adc {
         // Setup first filter configuration register. 10Hz data rate. Sinc5Sinc1 Filter. No postfilter.
         adc.write(
             ad7172::AdcReg::FILTCON0,
-            ad7172::Filtcon::Order::SINC5SINC1 | ad7172::Filtcon::Odr::ODR_10,
+            ad7172::Filtcon::Order::SINC5SINC1 | ad7172::Filtcon::Odr::ODR_1_25,
         );
+
+        // Re-apply (also set after ADC reset) SYNC_EN flag in gpio register for standard synchronization
+        adc.write(ad7172::AdcReg::GPIOCON, ad7172::Gpiocon::SyncEn::ENABLED);
+    }
+
+    /// Handle adc interrupt.
+    pub fn handle_interrupt(&mut self) -> (InputChannel, u32) {
+        let (current_phy, ch) = Self::SCHEDULE[self.current_position];
+
+        let (data, status) = self.adcs.read_data();
+
+        self.rdyn.clear_interrupt_pending_bit();
+
+        set_cs!(self, current_phy, High);
+
+        self.current_position = (self.current_position + 1) % Self::SCHEDULE.len();
+
+        let (current_phy, _) = Self::SCHEDULE[self.current_position];
+
+        set_cs!(self, current_phy, Low);
+
+        assert_eq!(status & 0x3, ch as u8 & 1); // check if correct input channel
+
+        (ch, data) // data as °C
+    }
+
+    /// Initiate the sampling sequence.
+    pub fn initiate_sampling(&mut self) {
```

Comment:
Please inline this. What's the point?
It doesn't seem to initiate anything. It just sets a single CS.
The sampling on all ADCs has already been initiated.
And it is broken to call it (`pub`) twice.

Reply:
I would have to export the macro to be able to start at a specific index. Do you think its fine to just select the first ADC and assuming we are at schedule index 0?

Reply:
Refactored.